CMakeLists.txt

```diff
@@ -42,6 +42,9 @@ if (MSVC AND EDGE_SDK_MSVC_PARALLEL_BUILD_ENABLE)
     message(STATUS "MSVC - Using parallel compilation (/MP)")
 endif (MSVC AND EDGE_SDK_MSVC_PARALLEL_BUILD_ENABLE)
 
+# Add preprocessor definitions
+add_compile_definitions(EDGE_SDK_VERSION_STRING="${olp-cpp-sdk_VERSION}")
```

Comment:
I would prefer to add individual numbers as defines.
```
add_compile_definitions(EDGE_SDK_VERSION_STRING="${olp-cpp-sdk_VERSION}")
add_compile_definitions(EDGE_SDK_VERSION_MAJOR="${olp-cpp-sdk_VERSION_MAJOR}")
add_compile_definitions(EDGE_SDK_VERSION_MINOR="${olp-cpp-sdk_VERSION_MINOR}")
add_compile_definitions(EDGE_SDK_VERSION_PATCH="${olp-cpp-sdk_VERSION_PATCH}")
```

Reply:
Done, but used actual integers instead strings for Major / Minor / Patch versions (in order to perform versions check)

Maybe we should add the version integer in the way it boost does:
```
//  BOOST_VERSION % 100 is the patch level
//  BOOST_VERSION / 100 % 1000 is the minor version
//  BOOST_VERSION / 100000 is the major version

#define BOOST_VERSION 104500
```